src/System.Drawing.Common/src/System/Drawing/GdiplusNative.Unix.cs

```diff
@@ -0,0 +1,2508 @@
+﻿using System.Drawing.Drawing2D;
+using System.Drawing.Imaging;
+using System.Drawing.Text;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+
+namespace System.Drawing
+{
+    internal partial class SafeNativeMethods
+    {
+        internal unsafe partial class Gdip
+        {
+            private const string LibName = "libdl";
+
+            public const int RTLD_NOW = 0x002;
+
+            [DllImport(LibName)]
+            public static extern IntPtr dlopen(string fileName, int flags);
+
+            [DllImport(LibName)]
+            public static extern IntPtr dlsym(IntPtr handle, string name);
+
+            [DllImport(LibName)]
+            public static extern int dlclose(IntPtr handle);
+
+            [DllImport(LibName)]
+            public static extern string dlerror();
```

Comment:
This P/Invoke for dlerror looks wrong.  Does dlerror allocate the char* it returns and expect the caller to free it with `free`?  If it does, this is fine, but assuming not, this is going to cause problems, as the runtime will assume it owns the returned char* and try to free it as part of marshaling it to string, in which case it should instead be defined to return `char*` or `IntPtr` and Marshal.PtrToStringAnsi or Marshal.PtrToStringUTF8 should be used to get the resulting string value.